### PR TITLE
Kanban Phase 2 (Part 1)

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Potential features (May split into multiple PR based on complexity of changes):
- [ ] Try to modernize the Kanban JS code now that IE is dropped and we can use ES 6 (Ongoing change)
- [x] Preview card content without having to open the card's form
- [x] Open card's main form in a dialog for editing instead of navigating away from the Kanban
- [ ] ~~Assign user from Kanban~~ Deferred for another PR
- [ ] ~~Maybe show subtasks as a checklist (Show them as we are currently with the option to expand?)~~ Deferred for another PR
- [ ] ~~When moving cards to a "finished" column prompt to set percent done to 100%. update sub-projects/sub-tasks accordingly. There needs to be a flag passed into the JS for those columns to support other itemtypes such as Tickets which have a solved and closed state.~~ Deferred for another PR
- [x] Add ability to delete cards and the underlying item

New features may be added to this list later.

This exists separate from the request for adding Kanban to ITIL Objects, so the features here may need replicated for those items if accepted.
I'm also trying to see if there are any changes I can make that reduce the amount of duplicated code in the tasklists plugin (I had a bad case of deja vu with that...) without replacing it. Better to expand on existing core code than trying to continuously copying it whenever there is a bug fix or feature.

New features will try to be added fully in a single commit, so it is best reviewed commit by commit.

### CODE REVIEW

**Comment by trasher** [comment]
@cconard96 could you please rebase and fix conflict?

File: js/kanban.js [hunk 1]
```diff
@@ -251,20 +322,34 @@
...
+         if (args[0]['rights'] !== undefined) {
+            self.rights = new GLPIKanbanRights(args[0]['rights']);
+         } else {
+            // 9.5.0 style compatibility
+            self.rights = new GLPIKanbanRights({
+               create_item: self.allow_add_item,
+               delete_item: self.allow_delete_item,
+               create_column: self.allow_create_column,
+               modify_view: self.allow_modify_view,
+               create_card_limited_columns: self.limit_addcard_columns
```
Comment (cedric-anne) [inline js/kanban.js:350]: There is still many occurences of usage of `allow_order_card` property.
```suggestion
               create_card_limited_columns: self.limit_addcard_columns,
               order_card: self.allow_order_card
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

File: js/kanban.js [hunk 2]
```diff
@@ -97,46 +192,22 @@
...
+      this.rights = new GLPIKanbanRights({});
 
-      /**
-       * If true, then a button will be added to the add column form that lets the user create a new column.
-       * For Projects as an example, it would create a new project state.
-       * Permissions are re-checked server-side during this request.
-       * @since 9.5.0
-       * @type {boolean}
-       */
+      /** @deprecated x.x.x Use rights.canCreateItem() instead */
```
Comment (trasher) [inline js/kanban.js:200]: Is this really required to keep that deprecated code? Does anything outside GLPI uses this object?

**Review by trasher — APPROVED** [review]

File: js/kanban.js (hunk 2 above)
Comment (cconard96) [reply js/kanban.js:200]: I don't know for sure. There could be a plugin out there that uses them. It probably isn't needed, but doesn't add much complexity leaving the old variables in.

Comment (trasher) [reply js/kanban.js:200]: OK, that was not a blocker anyways

File: ajax/kanban.php [hunk 3]
```diff
@@ -205,24 +213,44 @@
...
+   $item->getFromDB($_REQUEST['items_id']);
+   // Check if the item can be trashed and if the request isn't forcing deletion (purge)
+   $maybe_deleted = $item->maybeDeleted() && !($_REQUEST['force'] ?? false);
+   if (($maybe_deleted && $item->canDeleteItem()) || (!$maybe_deleted && $item->canPurgeItem())) {
+      $item->delete(['id' => $_REQUEST['items_id']], !$maybe_deleted);
+   } else {
+      http_response_code(403);
+      return;
+   }
+}
```
Comment (orthagh) [inline ajax/kanban.php:256]:
```suggestion
}

```

File: js/kanban.js [hunk 4]
```diff
@@ -699,36 +849,55 @@
...
+               $(self.element + ' #kanban-dialog').dialog({
+                  autoOpen: false,
+                  modal: true,
+                  resizable: true,
+                  draggable: true,
+                  height: 700,
+                  width: 800
+               });
+            }
+            $('#kanban-dialog').load((self.ajax_root + "kanban.php/?action=show_card_edit_form&itemtype="+itemtype+"&card=" + items_id)).dialog("open");
```
Comment (orthagh) [inline js/kanban.js:884]: this `/` at the end of the url do 404 errors on my current setup.

Is it mandatory ?
If i remove it, it seems to be ok on my side.

**Review by orthagh — APPROVED** [review]
apart the `/` issue -> ok for me
I still set an approved review (conditioned by a fix on the above point)

File: js/kanban.js (hunk 4 above)
Comment (cconard96) [reply js/kanban.js]: It should be optional to have it in a valid URL. Works for me with or without, so I removed the slash.

**Comment by orthagh** [comment]
ok, good for merge so

**Pull request merged**